hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v2/ActiveTimelineV2.java

```diff
@@ -727,17 +725,15 @@ public void createFileInMetaPath(String filename, Option<byte[]> content, boolea
   protected void createCompleteFileInMetaPath(boolean shouldLock, HoodieInstant instant, Option<byte[]> content) {
     TimeGenerator timeGenerator = TimeGenerators
         .getTimeGenerator(metaClient.getTimeGeneratorConfig(), metaClient.getStorageConf());
-    timeGenerator.consumeTime(!shouldLock, currentTimeMillis -> {
-      String completionTime = HoodieInstantTimeGenerator.formatDate(new Date(currentTimeMillis));
-      String fileName = instantFileNameGenerator.getFileName(completionTime, instant);
-      StoragePath fullPath = getInstantFileNamePath(fileName);
-      if (metaClient.getTimelineLayoutVersion().isNullVersion()) {
-        FileIOUtils.createFileInPath(metaClient.getStorage(), fullPath, content);
-      } else {
-        metaClient.getStorage().createImmutableFileInPath(fullPath, content.map(HoodieInstantWriter::convertByteArrayToWriter));
-      }
-      LOG.info("Created new file for toInstant ?" + fullPath);
-    });
+    String completionTime = TimelineUtils.generateInstantTime(shouldLock, timeGenerator);
```

Comment:
The lock scope can be narrowed, we only need to lock for time generation because the file creation is already atomic and the time is unique so there should be no conflicts.

Reply:
Consider this case:
```
                     String completionTime = "1000"                                file creation
thread 1             |                                                                                   |
thread 2                        |                                                          |
                                String completionTime = "1002"    file creation
```
This can let the file with smaller completion time written later in time.  Should this be avoided?

Reply:
Should be fine, the completion time is part of the file name which is used as the file completion time instead of the modification time.

Reply:
Sounds good.  I fixed it.